pkg/mutation/assign_mutator.go

```diff
@@ -39,6 +40,10 @@ func (m *AssignMutator) SchemaBindings() []SchemaBinding {
 	return m.bindings
 }
 
+func (m *AssignMutator) Value() runtime.RawExtension {
+	return m.assign.Spec.Parameters.Assign
```

Comment:
See https://github.com/open-policy-agent/gatekeeper/pull/937/files#r532601518

I think it makes sense to do a step further, and return assign["value"] as an interface{} that can then either be a map or a scalar.

Reply:
+1

Reply:
Also, we should cache the parsing of that value to avoid paying the serialization/deserialization penalty every time.

Reply:
Though if it's too hard to deepcopy that, it may be worth paying the penalty.

Reply:
Looks like Kubernetes has already dealt with the "deepcopy a JSON interface" problem:

https://github.com/kubernetes/apimachinery/blob/2456ebdaba229616fab2161a615148884b46644b/pkg/runtime/converter.go#L437

Reply:
Still, I don't see the need for deep copying. m.assign is already deepcopied (iirc), and the "json" (which actually is a map) will always be read only.
Am I missing something?

Reply:
Just paranoia that consumers won't start modifying the return value of `.Value()` in the future, which would create a pretty hard-to-debug bug.

The idea here is just to use deepcopy to return a completely separate value to the user so we don't need to worry (not super worried about the performance penalty of not copying a deserialization cache).

Reply:
Understood, and yes, I feel the pain of debugging that kind of bug :P 
And in any case, we can optimize afterwards, after a pass of profiler if needed.

Reply:
+1

Reply:
Sounds to me like we just deserialize every time we call Value()?

Reply:
iirc, the object is already deserialized in a map[string]interface{} form.
If we want to do the deepcopy, we could deepcopy assign and return copyOfAssign["value"].
In this way we would return a interface{} that can either be a map or a scalar.

Reply:
I think `Spec.Parameters.Assign` is of type [RawExtension](https://godoc.org/k8s.io/apimachinery/pkg/runtime#RawExtension), which stores the JSON serialized as type `[]byte`

I am on the same page as you with returning `copyOfAssign["value"]`, I think we just have slightly different understandings of what the types of these objects are.

Reply:
Yes, forget my last comment. The field is a raw json. 
Let's start deserializing each time, if not we may add a field containing the deserialized object.

Reply:
Deserializing each time for now.
Let' revisit the caching in a new PR if needed, and stick with an mvf for now.